docs/ansible/coding-conventions.md

```diff
@@ -0,0 +1,669 @@
+
+## Names
+
+All the newly created Ansible roles should follow the name convention using dashes if necessary:
+`[company]-[action]-[function/technology]`
+
+✅ ***Good***
+
+```yaml
+# good
+mycompany-setup-lvm
+```
+
+❌ ***Bad***
+```yaml
+# bad
+lvm
+```
+
+
+## Use Modules instead of command or shell
+Before using the `command` or `shell` module, verify if there is already a module available which can avoid the usage of raw shell command.
+
+✅ ***Good***
+
+```yaml
+# good
+- name: install packages
+  tasks:
+    - name: 'install httpd'
+      yum:
+        name: 'httpd'
+        state: 'present'
+```
+
+❌ ***Bad***
+```yaml
+# bad
+- name: install httpd
+  tasks:
+    - command: "yum install httpd"
+```
+
+**Reason:** While raw command could be seen as a security risk in general, another reason to avoid them is the loss of immutability of the ansible playbooks or roles. Ansible cannot verify if a command has been already executed before or not and will therefore execute it every time the playbook is running.
+
+## 🔴 Spacing addons
+
+# Comparing
+Do not compare to literal True/False.
+✅ ***Good***
+
+```yaml
+when: var
+```
+```yaml
+when: not var
+```
+
+❌ ***Bad***
+```yaml
+when: var == True
+```
+```yaml
+when: var == Yes
+```
+```yaml
+when: var != ""
+```
+
+## Playbook File Extension
+
+All Ansible Yaml files should have a .yml extension (and NOT .YML, .yaml etc).
+
+##Vaults
+
+All Ansible Vault files should have a .vault extension (and NOT .yml, .YML, .yaml etc).
+
+## Use Module synchronize Instead of copy for Large Files
+
+## Use Block-Module
+
+Block can help to organize the code and can enable rollbacks.
+```yaml
+- block:
+    copy:
+      src: critical.conf
+      dest: /etc/critical/crit.conf
+    service:
+      name: critical
+      state: restarted
+  rescue:
+   command: shutdown -h now
+```
+
+## Order in playbook
+
+Playbook definitions should follow this order.
+
+* `name`
+* `hosts`
+* `remote_user`
+* `become`
+* `vars`
+* `pre_tasks`
+* `roles`
+* `tasks`
+
+```yaml
+---
+- name: update root authorized_keys for all machines
+  hosts:
+    - all
+  become: true
+
+  vars:
+    root_keys_users:
+      - user: user1
+        key: |
+          ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIHkCVF05JvfkrfOOESivOxV4N8+A/EMEkF7/nCQMRoQg
+        enabled: true
+
+  pre_tasks:
+    - name: add custom root users
+      set_fact:
+        root_keys_users: "{{ root_keys_users | union( root_keys_users_custom|default([]) ) }}"
+
+  roles:
+    - role: root-keys
+
+  tasks:
+    - name: print out debug message
+      ansible.builtin.debug:
+        msg: "fee foo faa"
+```
+
+ Why do this?
+
+A common order makes playbooks consistent and easier to read for your dear colleagues. Think of them when you write.
+
+## Order in task declaration
+
+A task should be declared in this order.
+
+* `name`
+* module and the arguments for the module
+* arguments for the task in alphabetic order
```

Comment:
list common optional args in the correct order (because this part is tricky and complex to remember)

loop, when - very common and has to be included

Reply:
Not sure that all written above should be implied here